pkg/cluster/clustermsi.go

```diff
@@ -62,40 +54,12 @@ func (m *manager) ensureClusterMsiCertificate(ctx context.Context) error {
 		return err
 	}
 
-	now := time.Now()
-
-	var expirationDate time.Time
-	if m.env.FeatureIsSet(env.FeatureUseMockMsiRp) {
```

Comment:
I didn't quite understand why we handled this logic after getting data from the MSI dataplane in mock mode, since we control the mock entirely. It's simple enough to just do the right think in the mock response.